cellarium/ml/core/saving.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from typing import Any
+
+import lightning.pytorch as pl
+from jsonargparse import ArgumentParser, Namespace
+from lightning.pytorch.utilities.rank_zero import rank_zero_warn
+
+
+def _load_state(
+    cls: type[pl.LightningModule],
+    checkpoint: dict[str, Any],
+    strict: bool | None = None,
+    **cls_kwargs_new: Any,
+) -> pl.LightningModule:
+    """
+    Re-implementation of :func:`lightning.pytorch.core.saving._load_state` that instantiates the model
+    using the configuration saved in the checkpoint.
+    """
+    ### cellarium.ml - this part is different from the original
+    parser = ArgumentParser()
+    parser.add_class_arguments(cls, "model")
+
+    cfg = Namespace(model={})
+
+    # pass in the values we saved automatically
+    if cls.CHECKPOINT_HYPER_PARAMS_KEY in checkpoint:
+        cfg = checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY]
+
+    # update cfg with cls_kwargs_new, such that new has higher priority
+    for key, value in cls_kwargs_new.items():
+        cfg["model"][key] = value
+
+    # instantiate the model
+    cfg_init = parser.instantiate_classes(cfg)
+    obj = cfg_init.model
+
+    # save the cfg to the :attr:`obj.hparams` to be able to load the model checkpoint
+    obj._set_hparams(cfg)
```

Comment:
Added this to save the config back to the object so that hyperparams are saved if the loaded model is checkpointed again.